_packages/@karrotmarket/gatsby-theme-team-website/src/templates/FaqPage.tsx

```diff
@@ -30,35 +36,132 @@ export const query = graphql`
           text
         }
 
-        ...TeamWebsite_FaqAccordion_faqData
-        faq_entries {
-          question
-          answer {
-            text
+        faq_page_entries {
+          faq_category_title
+          faq_page {
+            id
+            uid
+            document {
+              ... on PrismicFaq {
+                data {
+                  display_name
+                }
+              }
+            }
           }
         }
       }
     }
+
+    prismicFaq(
+      id: { eq: $id }
+    ) {
+      uid
+      data {
+        ...TeamWebsite_FaqAccordion_faqData
+        entries {
+          keywords
+        }
+      }
+    }
   }
 `;
 
 const Container = styled('main', {
   contentArea: true,
-  maxWidth: rem(760),
+  maxWidth: rem(1200),
 });
 
 const PageTitle = styled(_PageTitle, {
-  marginBottom: rem(80),
+  marginBottom: rem(56),
 
   '@sm': {
-    marginBottom: rem(100),
+    marginBottom: rem(80),
   },
 });
 
+const Filters = styled('div', {
+  display: 'flex',
+  justifyContent: 'space-between',
+  marginBottom: rem(52),
+  flexDirection: 'column',
+
+  '@lg': {
+    flexDirection: 'row'
+  }
+});
+
+const FaqGroupWrapper = styled('ul', {
+  display: 'flex',
+  alignItems: 'center',
+  padding: '0',
+  overflow: 'auto',
+  gap: rem(50),
+  marginBottom: rem(52),
+
+  '@lg': {
+    marginBottom: '0'
+  }
+});
+
+const FaqGroup = styled('li', {
+  fontSize: '$subtitle3',
+  fontWeight: 'bold',
+  listStyle: 'none',
+  float: 'left',
+  cursor: 'pointer',
+  whiteSpace: 'nowrap',
+});
+
+const FaqGroupLink = styled(Link, {
+  color: '$gray900',
+  textDecoration: 'none',
+  variants: {
+    selected: {
+      true: {
+        color: '$carrot600',
+      },
+      false: {
+        '&:hover': {
+          color: '$gray600'
+        },
+      }
+    }
+  }
+})
+
+const FaqList = styled(_FaqList, {
+  minHeight: '80vh',
+});
+
+const Search = styled(_Search, {
+  '@lg': {
+    minWidth: rem(300),
+  }
+});
+
 const FaqPage: React.FC<FaqPageProps> = ({
   data,
+  location
 }) => {
-  required(data.prismicTeamContents?.data);
+  const messages = useTranslation();
+
+  const searchParams = new URLSearchParams(location.search);
```

Comment:
```suggestion
  const searchParams = new URLSearchParams(location.search);
```

이건 useSyncExternalStore 써서 동기화 및 서버 폴백 처리가 명시적으로 되어야 안전할 것 같습니다.

Reply:
오.. `useSyncExternalStore` 처음 알았네요ㅋㅋㅋ 👍